dask_sql/input_utils/pandaslike.py

```diff
@@ -18,6 +18,20 @@ def is_correct_input(
         is_cudf_type = cudf and isinstance(input_item, cudf.DataFrame)
         return is_cudf_type or isinstance(input_item, pd.DataFrame) or format == "dask"
 
-    def to_dc(self, input_item, table_name: str, format: str = None, **kwargs):
+    def to_dc(
+        self,
+        input_item,
+        table_name: str,
+        format: str = None,
+        gpu: bool = False,
+        **kwargs,
+    ):
         npartitions = kwargs.pop("npartitions", 1)
-        return dd.from_pandas(input_item, npartitions=npartitions, **kwargs)
+        if gpu:  # pragma: no cover
+            import dask_cudf
+
+            return dask_cudf.from_cudf(
+                cudf.from_pandas(input_item), npartitions=npartitions, **kwargs,
+            )
+        else:
+            return dd.from_pandas(input_item, npartitions=npartitions, **kwargs)
```

Comment:
Given that this input util accepts both `cudf` and `pandas` dataframes as valid inputs, you'd probably need an additional check here to check if `input_item` is a pandas dataframe or not, and call the `from_pandas` function only for that case.